lib/chef/provider/package/aix.rb

```diff
@@ -83,21 +96,14 @@ def load_current_resource
         end
 
         def candidate_version
-          return @candidate_version if @candidate_version
-          ret = shell_out_with_timeout("installp -L -d #{@new_resource.source}")
-          ret.stdout.each_line do |line|
-            case line
-            when /\w:#{Regexp.escape(@new_resource.package_name)}:(.*)/
-              fields = line.split(":")
-              @candidate_version = fields[2]
-              @new_resource.version(fields[2])
-              Chef::Log.debug("#{@new_resource} setting install candidate version to #{@candidate_version}")
+          @candidate_version ||= begin
+            candidate, fs = filter_source_device
```

Comment:
that's correct and the same as before. if no matching filesets are in source, candidate is not set